src/Controllers/Scheduler/listDemos.php

```diff
@@ -16,8 +16,20 @@
 $currentUser = MyRadio_User::getInstance();
 
 foreach ($demos as $demo) {
-    if ($currentUser->hasAuth(AUTH_ADDDEMOS)) {
+    if ($currentUser->hasAuth(AUTH_CANCELDEMOS)) {
         $demo['attending'] = MyRadio_Demo::usersAttendingDemo($demo['show_season_timeslot_id']);
+
+        if (MyRadio_Demo::isDemoEmpty($demo['show_season_timeslot_id'])) {
+            $demo['attending'] = 'Empty';
+            $demo = addCancelButton($demo);
+        }
+    } elseif ($currentUser->hasAuth(AUTH_ADDDEMOS)) {
+        $demo['attending'] = MyRadio_Demo::usersAttendingDemo($demo['show_season_timeslot_id']);
+
+        if (MyRadio_Demo::isDemoEmpty($demo['show_season_timeslot_id'] && MyRadio_Demo::getDemoer() == $currentUser)) {
```

Comment:
This line still needs the timeslot ID fed to getDemoer, leads to some very nice errors on dev :P
Probably better to shorten the whole $demo['xxx'] into a var on previous line in that case.